### PR TITLE
Added a test using $ as escape character.

### PR DESCRIPTION
**Pull request by henryso**
Updated expectations with minor spacing changes.
Updated gabc-gabc expectations due to $ as escape character.

Tests gregorio-project/gregorio#1515

### CODE REVIEW

File: tests/gabc-dump/glyphs/above_lines_text.gabc [hunk 1]
```diff
@@ -1,4 +1,4 @@
 name: above lines text;
 %%
-gh<alt>above lines</alt>(ghZ)
-gh(gh[alt:note alt]Z)
+gh<alt>above lines $(parentheses)</alt>(ghZ)
```
Comment (rpspringuel) [inline tests/gabc-dump/glyphs/above_lines_text.gabc:3]: Are the closing parentheses not escaped?

File: tests/gabc-dump/glyphs/protrusion.exp [hunk 2]
```diff
@@ -14,13 +14,15 @@ et(i) (::)
 et(i) (::Z)
 Nus:(i) *(,)
 et(i) (::)
-Dus<pr>:(i) *(,)
+Nus$:(i) *(,)
```
Comment (rpspringuel) [inline tests/gabc-dump/glyphs/protrusion.exp:17]: What's going on here?  The changed lines seem to be mismatched.

**Comment by rpspringuel** [comment]
@henryso  I've also submitted a PR against this branch in your fork of the repository.

File: tests/gabc-dump/glyphs/protrusion.exp (hunk 2 above)
Comment (henryso) [reply tests/gabc-dump/glyphs/protrusion.exp:17]: I added a case showing an escaped `:` as not having protrusion meaning.

File: tests/gabc-dump/glyphs/above_lines_text.gabc (hunk 1 above)
Comment (henryso) [reply tests/gabc-dump/glyphs/above_lines_text.gabc:3]: No, they are not.  Lexically, close parenthesis does not have special meaning in ordinary text.

File: tests/gabc-dump/glyphs/protrusion.exp (hunk 2 above)
Comment (rpspringuel) [reply tests/gabc-dump/glyphs/protrusion.exp:17]: So it's just that the added case is similar enough to an existing line that the diff tool gets confused when trying to match up the lines?

Comment (henryso) [reply tests/gabc-dump/glyphs/protrusion.exp:17]: Yes.

File: tests/gabc-dump/glyphs/above_lines_text.gabc [hunk 3]
```diff
@@ -1,4 +1,4 @@
 name: above lines text;
 %%
 gh<alt>above lines $(parentheses)</alt>(ghZ)
-gh$(with parentheses)(gh[alt:note alt]Z)
+gh$(with parentheses)(gh[alt:note [alt$]]Z)
```
Comment (rpspringuel) [inline tests/gabc-dump/glyphs/above_lines_text.gabc:4]: I'd make this into two lines as follows:

```
gh$(with parentheses)(gh[alt:note (alt$)]Z)
gh$(with brackets)(gh[alt:note [alt$]]Z)
```

**Comment by rpspringuel** [comment]
Oh, and don't forget the PR: henryso/gregorio-test#2

**Pull request merged**